tooling/debugger/src/context.rs

```diff
@@ -21,6 +21,117 @@ use thiserror::Error;
 use std::collections::BTreeMap;
 use std::collections::{hash_set::Iter, HashSet};
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub struct AddressMap {
+    // A Noir program is composed by
+    // `n` ACIR circuits
+    //       |_ `m` ACIR opcodes
+    //                |_ Acir call
+    //                |_ Acir Brillig function invocation
+    //                           |_ `p` Brillig opcodes
+    //
+    // The purpose of this structure is to map the opcode locations in ACIR circuits into
+    // a flat contiguous address space to be able to expose them to the DAP interface.
+    // In this address space, the ACIR circuits are laid out one after the other, and
+    // Brillig functions called from such circuits are expanded inline, replacing
+    // the `BrilligCall` ACIR opcode.
```

Comment:
Oh, you can move this above the `#[derive()]` annotation above so it applies to the whole structure. I think you can also use `///` (triple slashes) for the documentation system to pick it up.